python/pyarrow/gandiva.pyx

```diff
@@ -405,9 +437,24 @@ cpdef make_projector(Schema schema, children, MemoryPool pool):
                                 &result))
     return Projector.create(result, pool)
 
+cpdef make_projector_with_mode(Schema schema, children,
+                               str selection_mode, MemoryPool pool):
```

Comment:
Some duplicated code could be removed by making selection_mode an optional argument of make_projector instead of writing a new function